From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 28 Feb 2022 19:02:49 +0930
Subject: [PATCH] Prevent world data saving


diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/EntityStorage.java b/src/main/java/net/minecraft/world/level/chunk/storage/EntityStorage.java
index bee39dee1b96023c907407877aedf3aafaf5e1b8..c501af8ff7fd621efb2696ee68e369b9afd5d7ed 100644
--- a/src/main/java/net/minecraft/world/level/chunk/storage/EntityStorage.java
+++ b/src/main/java/net/minecraft/world/level/chunk/storage/EntityStorage.java
@@ -100,6 +100,7 @@ public class EntityStorage implements EntityPersistentStorage<Entity> {
         return saveEntityChunk0(entities, chunkPos, level, false);
     }
     private static CompoundTag saveEntityChunk0(List<Entity> entities, ChunkPos chunkPos, ServerLevel level, boolean force) {
+        if (true) return null; // NabArchive
         if (!force && entities.isEmpty()) {
             return null;
         }
diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index f6e3b745fc417354380d4a969f83aee430bad785..c74968927905abfb41adfe12b0a08254477f0256 100644
--- a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -299,6 +299,7 @@ public class RegionFileStorage implements AutoCloseable {
     }
 
     protected void write(ChunkPos pos, @Nullable CompoundTag nbt) throws IOException {
+        if (true) return; // NabArchive
         // Paper start - rewrite chunk system
         RegionFile regionfile = this.getRegionFile(pos, nbt == null, true); // CraftBukkit
         if (nbt == null && regionfile == null) {
diff --git a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
index 6e23e69abd56eeda3b52a22019e1b74ae10682e7..c3b9ef6049024ff7c2ea9b8353e8d80686c65f41 100644
--- a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -134,6 +134,7 @@ public class DimensionDataStorage implements java.io.Closeable { // Paper - Writ
     // Paper end - Write SavedData IO async
 
     public void save(boolean async) { // Paper - Write SavedData IO async
+        if (true) return; // NabArchive
         this.cache.forEach((id, state) -> {
             if (state != null) {
                 // Paper start - Write SavedData IO async
