From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 28 Feb 2022 21:07:01 +0930
Subject: [PATCH] Stop warnings about duplicate UUIDs


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 3b1a34b34979ab436ccd33f0a85bfae537cbecb4..1b51761decd1d0dc41213f0624491daf4c0954fa 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1388,7 +1388,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         org.spigotmc.AsyncCatcher.catchOp("entity add"); // Spigot
         // Paper start
         if (entity.valid) {
-            MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable());
+            //MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable()); // NabArchive
 
             if (DEBUG_ENTITIES) {
                 Throwable thr = entity.addedToWorldStack;
diff --git a/src/main/java/net/minecraft/world/level/entity/EntityLookup.java b/src/main/java/net/minecraft/world/level/entity/EntityLookup.java
index f4731d69946871768ecf16149a47d2ef1cb3e910..6c6d7ad77d14db195a1c1d4583ee2a952e7e6e85 100644
--- a/src/main/java/net/minecraft/world/level/entity/EntityLookup.java
+++ b/src/main/java/net/minecraft/world/level/entity/EntityLookup.java
@@ -33,7 +33,7 @@ public class EntityLookup<T extends EntityAccess> {
     public void add(T entity) {
         UUID uUID = entity.getUUID();
         if (this.byUuid.containsKey(uUID)) {
-            LOGGER.warn("Duplicate entity UUID {}: {}", uUID, entity);
+            //LOGGER.warn("Duplicate entity UUID {}: {}", uUID, entity); // NabArchive
             // Paper start - extra debug info
             if (entity instanceof net.minecraft.world.entity.Entity entityCast) {
                 if (net.minecraft.server.level.ServerLevel.DEBUG_ENTITIES) {
diff --git a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
index 3d5f38a1ca0206e5bcae4ed49d3df7ce0967f722..086c842efc043f135f3a169a5c0bfd520b262bc7 100644
--- a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -147,7 +147,7 @@ public class PersistentEntitySectionManager<T extends EntityAccess> implements A
                 return true;
             }
             // Paper end
-            PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity);
+            //PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity); // NabArchive
             // Paper start
             if (net.minecraft.world.level.Level.DEBUG_ENTITIES && ((Entity) entity).level.paperConfig.duplicateUUIDMode != com.destroystokyo.paper.PaperWorldConfig.DuplicateUUIDMode.NOTHING) {
                 if (((Entity) entity).addedToWorldStack != null) {
