From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 28 Feb 2022 21:07:01 +0930
Subject: [PATCH] Stop warnings about duplicate UUIDs


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 0981d440d0dbfe4df668d1f3f1b5706a93bc4434..894af8219c4f51946bc2c65f8f7f97814c2f50df 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1671,7 +1671,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         Entity entity = (Entity) this.getEntities().get(player.getUUID());
 
         if (entity != null) {
-            ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID());
+            //ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID()); // NabArchive
             entity.unRide();
             this.removePlayerImmediately((ServerPlayer) entity, Entity.RemovalReason.DISCARDED);
         }
@@ -1685,7 +1685,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         entity.generation = false; // Paper - Don't fire sync event during generation; Reset flag if it was added during a ServerLevel generation process
         // Paper start - extra debug info
         if (entity.valid) {
-            MinecraftServer.LOGGER.error("Attempted Double World add on {}", entity, new Throwable());
+            //MinecraftServer.LOGGER.error("Attempted Double World add on {}", entity, new Throwable()); // NabArchive
             return true;
         }
         // Paper end - extra debug info
diff --git a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
index 1c6e8438219f355274db4e0fa849cdd90648fbb4..befa37bb5dbfbd29029f7113f8d06b411a75d312 100644
--- a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -80,7 +80,7 @@ public class PersistentEntitySectionManager<T extends EntityAccess> implements A
     private boolean addEntityUuid(T entity) {
         org.spigotmc.AsyncCatcher.catchOp("Entity add by UUID"); // Paper
         if (!this.knownUuids.add(entity.getUUID())) {
-            PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity);
+            //PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity); // NabArchive
             return false;
         } else {
             return true;
