From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 28 Feb 2022 21:07:01 +0930
Subject: [PATCH] Stop warnings about duplicate UUIDs


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index b396e9b35f315db37ba070ad4baeec1f098d0cb8..248754cf2fc6a43717930ccba248ab2662233be4 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1427,7 +1427,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         Entity entity = (Entity) this.getEntities().get(player.getUUID());
 
         if (entity != null) {
-            ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID().toString());
+            //ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID().toString()); // NabArchive
             entity.unRide();
             this.removePlayerImmediately((ServerPlayer) entity, Entity.RemovalReason.DISCARDED);
         }
@@ -1440,7 +1440,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         org.spigotmc.AsyncCatcher.catchOp("entity add"); // Spigot
         // Paper start
         if (entity.valid) {
-            MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable());
+            //MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable()); // NabArchive
 
             if (DEBUG_ENTITIES) {
                 Throwable thr = entity.addedToWorldStack;
diff --git a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
index e738f058af577150445ac4ef7a0bcbf98e3ba52e..0363f5c46f6d523eaed40759fb18cbb7a0f64b01 100644
--- a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -79,7 +79,7 @@ public class PersistentEntitySectionManager<T extends EntityAccess> implements A
     private boolean addEntityUuid(T entity) {
         org.spigotmc.AsyncCatcher.catchOp("Entity add by UUID"); // Paper
         if (!this.knownUuids.add(entity.getUUID())) {
-            PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity);
+            //PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity); // NabArchive
             return false;
         } else {
             return true;
