From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 28 Feb 2022 21:07:01 +0930
Subject: [PATCH] Stop warnings about duplicate UUIDs


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 083349794d5ceb50322c5a645dd33fbfcc1c8155..8013a53ea69994192f6d0070f382c7c3b00ddb90 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1397,7 +1397,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         Entity entity = (Entity) this.getEntities().get(player.getUUID());
 
         if (entity != null) {
-            ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID().toString());
+            //ServerLevel.LOGGER.warn("Force-added player with duplicate UUID {}", player.getUUID().toString()); // NabArchive
             entity.unRide();
             this.removePlayerImmediately((ServerPlayer) entity, Entity.RemovalReason.DISCARDED);
         }
@@ -1410,7 +1410,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         org.spigotmc.AsyncCatcher.catchOp("entity add"); // Spigot
         // Paper start
         if (entity.valid) {
-            MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable());
+            //MinecraftServer.LOGGER.error("Attempted Double World add on " + entity, new Throwable()); // NabArchive
 
             if (DEBUG_ENTITIES) {
                 Throwable thr = entity.addedToWorldStack;
diff --git a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
index 30c4974b4019d67cffabd3e686c782659def3ba6..75f3340cdf65d9de9d1b2b69ffa12364a6ce3853 100644
--- a/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/src/main/java/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -147,7 +147,7 @@ public class PersistentEntitySectionManager<T extends EntityAccess> implements A
                 return true;
             }
             // Paper end
-            PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity);
+            // PersistentEntitySectionManager.LOGGER.warn("UUID of added entity already exists: {}", entity); // NabArchive
             // Paper start
             if (net.minecraft.world.level.Level.DEBUG_ENTITIES && ((Entity) entity).level.paperConfig().entities.spawning.duplicateUuid.mode != io.papermc.paper.configuration.WorldConfiguration.Entities.Spawning.DuplicateUUID.DuplicateUUIDMode.NOTHING) {
                 if (((Entity) entity).addedToWorldStack != null) {
