From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Sun, 25 Sep 2022 20:07:28 +0930
Subject: [PATCH] Use map data from the current level if available


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 894af8219c4f51946bc2c65f8f7f97814c2f50df..41943ab8df02d9097f11ec9a90635b4c17b0909e 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -2137,7 +2137,8 @@ public class ServerLevel extends Level implements WorldGenLevel {
 
         final net.minecraft.world.level.saveddata.SavedData existing = storage.cache.get(id.key());
         if (existing == null && !storage.cache.containsKey(id.key())) {
-            final MapItemSavedData worldmap = (MapItemSavedData) this.getServer().overworld().getDataStorage().get(MapItemSavedData.factory(), id.key());
+            ServerLevel levelToUse = this.getDataStorage().dataExists(id.key()) ? this : this.getServer().overworld(); // NabArchive
+            final MapItemSavedData worldmap = (MapItemSavedData) levelToUse.getDataStorage().get(MapItemSavedData.factory(), id.key()); // NabArchive
             storage.cache.put(id.key(), worldmap);
             if (worldmap != null) {
                 worldmap.id = id;
diff --git a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
index c3b9ef6049024ff7c2ea9b8353e8d80686c65f41..f41c1f30be18552f6e24170defcf06264b841304 100644
--- a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -38,6 +38,14 @@ public class DimensionDataStorage implements java.io.Closeable { // Paper - Writ
         this.ioExecutor = java.util.concurrent.Executors.newSingleThreadExecutor(new com.google.common.util.concurrent.ThreadFactoryBuilder().setNameFormat("DimensionDataIO - " + dataFolder.getParent() + " - %d").setDaemon(true).build()); // Paper - Write SavedData IO async
     }
 
+    // NabArchive start
+    public boolean dataExists(String id) {
+        try {
+            return getDataFile(id).exists();
+        } catch (Exception ex) {return false;}
+    }
+    // NabArchive end
+
     private File getDataFile(String id) {
         return new File(this.dataFolder, id + ".dat");
     }
